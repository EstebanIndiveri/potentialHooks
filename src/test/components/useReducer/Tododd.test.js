import React from 'react';
import { shallow } from 'enzyme';
import Tododd from '../../../components/useReducer/TodoAdd';
import '@testing-library/jest-dom';
    describe('<Tododd/>', () => {
        const handleAddTodo=jest.fn();
        const wrapper=shallow(
            <Tododd
            handleAddTodo={handleAddTodo}
            />
            )
        test('match con snap', () => {
           expect(wrapper).toMatchSnapshot();
        });

        test('No debe de llamar handleAddTodo', () => {
            const formSumbit=wrapper.find('form').prop('onSubmit');
            formSumbit({preventDefault(){}});
            expect(handleAddTodo).toHaveBeenCalledTimes(0);
        });
        test('llama function handleAddTodo', () => {
            //argumento
            const value='apdender react';
            wrapper.find('input').simulate('change',{
                target:{
                    value,
                    name:'description'
                }
            });
            const formSumbit=wrapper.find('form').prop('onSubmit');
            formSumbit({preventDefault(){}});
            expect(handleAddTodo).toHaveBeenCalledTimes(1);
            expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object));
            expect(handleAddTodo).toHaveBeenCalledWith({
                id:expect.any(Number),
                desc:value,
                done:false,
            }); 
            expect(wrapper.find('input').prop('value')).toBe('');
        });
    })
    