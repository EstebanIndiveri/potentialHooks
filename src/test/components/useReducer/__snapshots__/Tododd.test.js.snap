// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tododd/> match con snap 1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Comenzar"
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-secondary mt-1 btn-block"
      type="submit"
    >
      Agregar
    </button>
  </form>
</Fragment>
`;
